indra/llappearanceutility/llprocesstexture.cpp

```diff
@@ -0,0 +1,285 @@
+
+    // Decompress the J2C image into a raw image.
+    LLPointer<LLImageRaw> image_raw = new LLImageRaw(j2c->getWidth(),
+                                                     j2c->getHeight(),
+                                                     j2c->getComponents());
+    const F32 MAX_DECODE_TIME = 60.f;
+    j2c->setDiscardLevel(0);
+    if (!j2c->decode(image_raw, MAX_DECODE_TIME))
+    {
+        throw LLAppException(RV_UNABLE_TO_DECODE, " Decoding timeout.");
+    }
+    // *TODO: Check for decode failure?
+    LL_DEBUGS() << "ID: " << id << ", Raw Width: " << image_raw->getWidth() << ", Raw Height: " << image_raw->getHeight()
+        << ", Raw Components: " << (S32) image_raw->getComponents()
+        << ", J2C Width: " << j2c->getWidth() << ", J2C Height: " << j2c->getHeight()
+        << ", J2C Components: " << (S32) j2c->getComponents() << LL_ENDL;
+
+    return image_raw;
+}
+
+void LLProcessTexture::parseInput(std::istream& input)
+{
+    mInputRaw = &input;
+    LL_DEBUGS() << "Reading..." << LL_ENDL;
+    // Parse LLSD header.
+    mInputData = LLSDSerialize::fromBinary( *mInputRaw, MAX_SIZE_LLSD_HEADER );
+    if (mInputData.isUndefined()) throw LLAppException(RV_UNABLE_TO_PARSE);
+    if (!mInputData.has("slot_id")) throw LLAppException(RV_UNABLE_TO_PARSE, " Missing slot id");
+    if (!mInputData.has("textures")) throw LLAppException(RV_UNABLE_TO_PARSE, " Missing texture header");
+    if (!mInputData.has("wearables")) throw LLAppException(RV_UNABLE_TO_PARSE, " Missing wearables");
+
+    // Verify the slot_id is valid.
+    if (BAKED_NUM_INDICES == LLAvatarAppearance::getDictionary()->findBakedByImageName(mInputData["slot_id"].asString()))
+    {
+        throw LLAppException(RV_UNABLE_TO_PARSE, " Invalid slot id");
+    }
+}
+
+void LLProcessTexture::init()
+{
+    if (mApp->isDebugMode()) gDebugGL = TRUE;
+
+    S32 maxTextureDecodedWidth = 0;
+    S32 maxTextureDecodedHeight = 0;
+
+    // Extract texture data.
+    LLSD::array_const_iterator iter = mInputData["textures"].beginArray();
+    LLSD::array_const_iterator end = mInputData["textures"].endArray();
+    std::map< LLUUID, LLPointer<LLImageRaw> > imageRawMap;
+
+    for (; iter != end; ++iter)
+    {
+        LLUUID texture_id = (*iter)[0].asUUID();
+        S32 texture_size = (*iter)[1];
+        imageRawMap[texture_id] = create_texture_from_stream(*mInputRaw, texture_size, texture_id);
+        maxTextureDecodedWidth = imageRawMap[texture_id]->getWidth() > maxTextureDecodedWidth ? imageRawMap[texture_id]->getWidth() : maxTextureDecodedWidth;
+        maxTextureDecodedHeight = imageRawMap[texture_id]->getHeight() > maxTextureDecodedHeight ? imageRawMap[texture_id]->getHeight() : maxTextureDecodedHeight;
+    }
+
+    if ("eyes" == mInputData["slot_id"].asString())
+    {
+        mBakeSize = 512;
+    }
+    else
+    {
+        // optimization: maybe we could use maxTextureDecodedWidth and 
+        //   maxTextureDecodedHeight for new LLBakingWindow below?
```

Comment:
Maybe we could use `maxTextureDecodedWidth` and `maxTextureDecodedHeight` for the `LLBakingWindow` below?